### PR TITLE
[TypeDeclaration] Add class method param type resolving by property

### LINKED ISSUE #2887: infer types of getters/setters from properties
would be great if rector could infer the param-type of a setter and the return-type of a getter

before
```php
<?php declare(strict_types=1);

final class DemoFile
{
	/**
	 * @var int
	 */
	private $id;
	
	public function setId($id) {
		$this->id = $id;
	}
	public function getId() {
		return $this->id;
	}
}
```

after
```php
<?php declare(strict_types=1);

final class DemoFile
{
	/**
	 * @var int
	 */
	private $id;
	
	public function setId(int $id) {
		$this->id = $id;
	}
	public function getId():int {
		return $this->id;
	}
}
```

### PR DESCRIPTION
**Pull request by TomasVotruba**
Closes #2887

### CODE REVIEW

**Comment by TomasVotruba** [comment]
cc @staabm

File: rules/type-declaration/tests/Rector/FunctionLike/ParamTypeDeclarationRector/Fixture/infer_from_property_type.php.inc [hunk 1]
```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace Rector\TypeDeclaration\Tests\Rector\ClassMethod\ParamTypeDeclarationRector\Fixture;
+
+class InferFromPropertyType
```
Comment (staabm) [inline rules/type-declaration/tests/Rector/FunctionLike/ParamTypeDeclarationRector/Fixture/infer_from_property_type.php.inc]:
```suggestion
class InferFromPropertyPhpdocType
```

File: rules/type-declaration/tests/Rector/FunctionLike/ParamTypeDeclarationRector/Fixture/infer_from_property_type.php.inc [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    }
+}
+
+?>
+-----
+<?php
+
+namespace Rector\TypeDeclaration\Tests\Rector\ClassMethod\ParamTypeDeclarationRector\Fixture;
+
+class InferFromPropertyType
```
Comment (staabm) [inline rules/type-declaration/tests/Rector/FunctionLike/ParamTypeDeclarationRector/Fixture/infer_from_property_type.php.inc]:
```suggestion
class InferFromPropertyPhpdocType
```

**Comment by staabm** [comment]
Maybe its a different topic, but I would also add the following example (Same idea but with typed property)

Before

```
class InferFromPropertyType
 {
     private int $id;

     public function setId($id)
     {
         $this->id = $id;
     }
 }
```

After

```
class InferFromPropertyType
 {
     private int $id;

     public function setId(int $id)
     {
         $this->id = $id;
     }
 }
```

**Comment by TomasVotruba** [comment]
> Thx for working on it. The test case shows what I expected. Cannot judge the implementation... not much AST experience on my end

Thanks, any review and feedback helps me :+1: . I'm blinded by my own ego.

**Comment by TomasVotruba** [comment]
> Maybe its a different topic, but I would also add the following example (Same idea but with typed property)

Good idea, I'll add new type inferer.

**Pull request merged**